cli/azd/pkg/containerapps/container_app.go

```diff
@@ -256,6 +285,155 @@ func (cas *containerAppService) AddRevision(
+	}
+
+	// Add a policy to capture the raw response. This allows us to get fields (ex: 'properties.runningStateDetails') that are
+	// currently not available in armappcontainers.ContainerAppsRevisionsClientGetRevisionResponse
+	responseCapture := azsdk.ResponseCapturePolicy{}
+
+	revisionsClientOptions := *cas.armClientOptions
+	revisionsClientOptions.PerCallPolicies = append(revisionsClientOptions.PerCallPolicies, &responseCapture)
+	revisionsClient, err := armappcontainers.NewContainerAppsRevisionsClient(
+		subscriptionId, credential, &revisionsClientOptions)
+	if err != nil {
+		return fmt.Errorf("creating revisions client: %w", err)
+	}
+
+	replicasClient, err := armappcontainers.NewContainerAppsRevisionReplicasClient(
+		subscriptionId, credential, cas.armClientOptions)
+	if err != nil {
+		return fmt.Errorf("creating replicas client: %w", err)
+	}
+
+	// Poll for the revision to be in a running state
+	// See https://learn.microsoft.com/en-us/azure/container-apps/revisions#lifecycle
+	// for the complete lifecycle of a revision
+	prevStatus := statusPrefix
+	delay := 3 * time.Second
+	pollCount := 0
+	for {
+		revision, err := revisionsClient.GetRevision(ctx, resourceGroupName, appName, revisionName, nil)
+		if err != nil {
+			return fmt.Errorf("getting revision '%s': %w", revisionName, err)
+		}
+
+		if !*revision.Properties.Active {
+			return fmt.Errorf("revision '%s' is not active", revisionName)
+		}
+
+		switch *revision.Properties.RunningState {
+		// Failed states
+		case armappcontainers.RevisionRunningStateFailed,
+			armappcontainers.RevisionRunningStateStopped,
+			armappcontainers.RevisionRunningStateDegraded:
+			errSuffix := ""
+
+			responseExtended := containerAppGetRevisionResponseExtended{}
+			if err := runtime.UnmarshalAsJSON(responseCapture.Response, &responseExtended); err == nil {
+				if responseExtended.Properties.RunningStateDetails != "" {
+					errSuffix = fmt.Sprintf(", %s", responseExtended.Properties.RunningStateDetails)
+				}
+			}
+
+			revisionManagementUrl := output.WithLinkFormat(
+				"%s/#@/resource/subscriptions/%s/resourceGroups/%s/providers/Microsoft.App/containerApps/%s/revisionManagement",
+				string(cas.portalUrlBase),
+				subscriptionId,
+				resourceGroupName,
+				appName)
+
+			if v, err := strconv.ParseBool(os.Getenv("AZD_DEMO_MODE")); err == nil && v {
+				revisionManagementUrl = fmt.Sprintf("Revision Management for %s in Azure Portal", appName)
+			}
+
+			return &internal.ErrorWithSuggestion{
+				Err: fmt.Errorf("revision '%s' is in a %s state%s",
+					revisionName,
+					*revision.Properties.RunningState,
+					errSuffix),
+				Suggestion: "To view logs:" +
+					"\n1. Visit " + revisionManagementUrl +
+					"\n2. Click on revision " + fmt.Sprintf("'%s'", revisionName) +
+					"\n3. View console and system logs" +
+					"\nFor more troubleshooting information, visit " +
+					output.WithLinkFormat("https://learn.microsoft.com/en-us/azure/container-apps/troubleshooting"),
```

Comment:
```suggestion
					output.WithLinkFormat("https://learn.microsoft.com/azure/container-apps/troubleshooting"),
```

NIT: Remove the en-us

This is a great error message by the way!